src/generate/register.rs

```diff
@@ -652,19 +656,80 @@ pub fn fields(
             }
 
             if width == 1 {
-                proxy_items.extend(quote! {
-                    ///Sets the field bit
-                    #inline
-                    pub #unsafety fn set_bit(self) -> &'a mut W {
-                        self.bit(true)
-                    }
+                if mwv != ModifiedWriteValues::Modify {
+                    unsafety = Some(Ident::new("unsafe", Span::call_site()));
+                }
 
-                    ///Clears the field bit
-                    #inline
-                    pub #unsafety fn clear_bit(self) -> &'a mut W {
-                        self.bit(false)
+                match mwv {
+                    ModifiedWriteValues::Modify => {
+                        proxy_items.extend(quote! {
+                            ///Sets the field bit
+                            #inline
+                            pub fn set_bit(self) -> &'a mut W {
+                                self.bit(true)
+                            }
+                            ///Clears the field bit
+                            #inline
+                            pub fn clear_bit(self) -> &'a mut W {
+                                self.bit(false)
+                            }
+                        });
                     }
-                });
+                    ModifiedWriteValues::OneToSet | ModifiedWriteValues::Set => {
+                        proxy_items.extend(quote! {
+                            ///Sets the field bit
+                            #inline
+                            pub fn set_bit(self) -> &'a mut W {
+                                unsafe { self.bit(true) }
+                            }
+                        });
+                    }
+                    ModifiedWriteValues::ZeroToClear | ModifiedWriteValues::Clear => {
+                        proxy_items.extend(quote! {
+                            ///Clears the field bit
+                            #inline
+                            pub fn clear_bit(self) -> &'a mut W {
+                                unsafe { self.bit(false) }
+                            }
+                        });
+                    }
+                    ModifiedWriteValues::OneToClear => {
+                        proxy_items.extend(quote! {
+                            ///Clears the field bit by passing one
+                            #inline
+                            pub fn clear_bit_by_one(self) -> &'a mut W {
+                                unsafe { self.bit(true) }
+                            }
+                        });
+                    }
+                    ModifiedWriteValues::ZeroToSet => {
+                        proxy_items.extend(quote! {
+                            ///Sets the field bit by passing zero
+                            #inline
+                            pub fn set_bit_by_zero(self) -> &'a mut W {
+                                unsafe { self.bit(false) }
```

Comment:
For `ZeroToSet`, how can the user set a `1` to _not_ set the bit, if only `set_bit_by_zero()` exists, if the reset value (for `write()`) or current value (for `modify`) is currently 0 already?

Same thing with `ZeroToToggle` below.